control.go

```diff
@@ -273,18 +279,21 @@ type connHost struct {
 }
 
 func (c *controlConn) setupConn(conn *Conn) error {
-	if err := c.registerEvents(conn); err != nil {
-		conn.Close()
-		return err
-	}
-
 	// TODO(zariel): do we need to fetch host info everytime
 	// the control conn connects? Surely we have it cached?
 	host, err := conn.localHostInfo(context.TODO())
 	if err != nil {
 		return err
 	}
 
+	if c.session.cfg.filterHost(host) {
```

Comment:
Shouldn't we filter-out the host before we dial the connection?

Reply:
I see, thanks! It seems we need to update the comment

```go
	// TODO(zariel): do we need to fetch host info everytime
	// the control conn connects? Surely we have it cached?
```

above as we now need to have fresh info for this use case every time, we can't just look it up by connect address anymore.

Reply:
I updated that comment to reflect that we need up-to-date info now for this call

Reply:
Thanks!